### PR TITLE
FIX: Fix double division of logistic model regularization by sum of weights

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR fixes an issue in which the regularization of logistic regression models passed to oneDAL is incorrect due to being divided by the sum of weights (or number of rows when no weights are present) twice when used from `sklearnex.linear_model.LogisticRegression`.

Note: I am not sure here if the function being modified is otherwise also accessible from some daal4py class or similar or if this might break something in daal4py outside of sklearnex.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/linear_model/tests/test_logreg.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 from numpy.testing import assert_allclose, assert_array_equal
 from scipy.sparse import csr_matrix
 from sklearn.datasets import load_breast_cancer, load_iris, make_classification
-from sklearn.metrics import accuracy_score
+from sklearn.linear_model import LogisticRegression as origLogisticRegression
```
Comment (icfaust) [inline sklearnex/linear_model/tests/test_logreg.py]:
```suggestion
from sklearn.linear_model import LogisticRegression as _sklearn_LogisticRegression

```
I'm not certain, just worried about how this interacts with sklearn's estimator discovery (```all_estimators```) which would see this as public, and would require an underscore (may be fine that its in a test file, not sure).

File: sklearnex/linear_model/tests/test_logreg.py [hunk 2]
```diff
@@ -132,3 +133,31 @@ def test_csr(queue, dtype, dims):
         assert_allclose(prob, prob_sp, rtol=rtol)
         assert_allclose(model.coef_, model_sp.coef_, rtol=rtol)
         assert_allclose(model.intercept_, model_sp.intercept_, rtol=rtol)
+
+
+def test_logistic_regression_is_correct():
```
Comment (icfaust) [inline sklearnex/linear_model/tests/test_logreg.py:145]: This is technically a conformance test. Thoughts @ethanglaser ?

File: sklearnex/linear_model/tests/test_logreg.py (hunk 1 above)
Comment (david-cortes-intel) [reply sklearnex/linear_model/tests/test_logreg.py]: Renamed.

File: sklearnex/linear_model/tests/test_logreg.py (hunk 2 above)
Comment (david-cortes-intel) [reply sklearnex/linear_model/tests/test_logreg.py:145]: It is taken from one of the conformance tests that was failing, but since we don't expect the test to pass due to the way that sklearnex approaches logistic regression, it tests the function value instead (oneDAL's result is oftentimes better than sklearn's).

Comment (icfaust) [reply sklearnex/linear_model/tests/test_logreg.py:145]: Ahhhh, interesting.  Tbh, I am not opposed to bringing it in as it is, though if you could comment to the related sklearn conformance test that would be good.

Comment (david-cortes-intel) [reply sklearnex/linear_model/tests/test_logreg.py:145]: Added a comment about it.

Comment (ethanglaser) [reply sklearnex/linear_model/tests/test_logreg.py:145]: In the event that we have a meaningful conformance test that must be deselected due to some numerical/minor differences in approach I think its a good idea to add a variant to our own testing
